### PR TITLE
output/reference: Include reference information in alert (if configured)

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11284   

When configured, include the reference value in the alert. The configuration value is in the `alert` section:  types.alert.reference. The default value is off/no. Set to yes to include the expanded reference from the rule in the alert record.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [4974](https://redmine.openinfosecfoundation.org/issues/4974)

Describe changes:
- Add `reference` value to suricata.yaml.in (default no/off)
- Set flag in output logger if the config setting is on
- Format the reference as a sequence, e.g., `references: [ "ref-1" [, "ref-2" [, ...]]]`

Updates: 
- Moved config setting to metadata.rule
- Rebase

### Provide values to any of the below to override the defaults.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1808

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
Noticed a couple of things with the `suricata.yaml.in` changes. The rest, as far as I could see, looks good :)

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21494

File: doc/userguide/output/eve/eve-json-output.rst [hunk 1]
```diff
@@ -70,6 +70,7 @@ Metadata::
             #payload-printable: yes   # enable dumping payload in printable (lossy) format
             #payload-length: yes      # enable dumping payload length, including the gaps
             #packet: yes              # enable dumping of packet (without stream segments)
+            #reference: no            # include reference information, if present, in the alert
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply doc/userguide/output/eve/eve-json-output.rst:73]: Yes, i'll make an update.

File: suricata.yaml.in [hunk 2]
```diff
@@ -167,6 +167,19 @@ outputs:
             # payload-length: yes      # enable dumping payload length, including the gaps
             # packet: yes              # enable dumping of packet (without stream segments)
             # metadata: no             # enable inclusion of app layer metadata with alert. Default yes
+            # If you want metadata, use:
+            # metadata:
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply suricata.yaml.in:171]: I think so as `metadata` is both a scalar value and a section.

**Comment by jlucovsky** [comment]
Continued in #11491

**Pull request closed without merging**